processor/src/decoder/mod.rs

```diff
@@ -413,7 +413,11 @@ impl Decoder {
 
     /// Get operation at a particular clock cycle. Only applicable in debug mode.
     pub fn get_operation_at(&self, clk: usize) -> Operation {
-        self.operations[clk]
+        if clk == 0 {
+            Operation::None
+        } else {
+            self.operations[clk - 1]
+        }
```

Comment:
I wouldn't put this logic into the decoder. Let's put it instead into the `VmStateIterator`.